python/hail/methods/statgen.py

```diff
@@ -1466,46 +1465,60 @@ def pca(entry_expr, k=10, compute_loadings=False, as_array=False):
     loadings = from_option(r._3())
     if loadings:
         loadings = Table(loadings)
-    return (jiterable_to_list(r._1()), scores, loadings)
+    return jiterable_to_list(r._1()), scores, loadings
 
 
-@typecheck(dataset=MatrixTable,
-           k=int,
+@typecheck(mt=MatrixTable,
            maf=numeric,
-           block_size=int,
+           k=nullable(int),
+           scores=nullable(Table),
            min_kinship=numeric,
-           statistics=enumeration("phi", "phik2", "phik2k0", "all"))
-def pc_relate(dataset, k, maf, block_size=512, min_kinship=-float("inf"), statistics="all"):
-    """Compute relatedness estimates between individuals using a variant
-    of the PC-Relate method.
+           statistics=enumeration('phi', 'phik2', 'phik2k0', 'all'),
+           block_size=nullable(int))
+def pc_relate(mt, maf, k=None, scores=None, min_kinship=-float("inf"), statistics="all", block_size=None):
```

Comment:
if k and scores are mutually exclusive arguments, make them keyword only arguments:
```
def pc_relate(mt, maf, min_kinship=-float("inf"), statistics="all", block_size=None, *, k=None, scores=None): ```

Reply:
@tpoterba what's the pythonista feeling on one argument that takes either an integer or a table? We'd need a new `either` type checker. I kind of like:

```
pc_relate(mt, mt.aaf, scores=5)
pc_realte(mt, mt.aaf, scores=mt.pc_score)
```

Reply:
Erm I guess I don't want a table, I really want a table-like expression. Unrelated issue though.

Reply:
That seems OK too. Won't `oneof` work here?

Reply:
Hrm, I'd suggest against. The user really should really be explicit about what they're putting in there.

Reply:
Also `scores=5` doesn't really make sense. It really is `k`

Reply:
langs really need to do a better job supporting mutually exclusive arguments, imo.

Reply:
No argument (heh) there. But I don't think it's so bad if they're documented next to each other, and as mutually exclusive.

Reply:
So I'll just make the `*, k=None, scores=None` change to force keywords?

Reply:
Actually, I prefer this, keeping the more commonly used parameters toward the front:
`pc_relate(mt, maf, *, k=None, scores=None, min_kinship=-float("inf"), statistics="all", block_size=None):`